source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerUITests/ADCIOSVisualizerUITests.mm

```diff
@@ -32,6 +38,156 @@ - (void)tearDown
     [super tearDown];
 }
 
+- (void)resetTestEnvironment
+{
+    XCUIElementQuery* buttons = testApp.buttons;
+    
+    // try to find Back button and tap it while it appears
+    XCUIElement* backButton = buttons[@"Back"];
+    while ([backButton exists])
```

Comment:
nit: this line worries me a little bit.
can you add a limit?
i think the max depth is 2. e.g 1.0/elements.
so maybe put 3 as a limit?